module/zfs/zvol.c

```diff
@@ -368,8 +368,18 @@ uzfs_ioc_stats(zfs_cmd_t *zc, nvlist_t *nvl)
 		if (zc->zc_name[0] == '\0' ||
 		    (uzfs_zvol_name_compare(zv, zc->zc_name) == 0)) {
 			nvlist_t *innvl = fnvlist_alloc();
+			/*
+			 * split the string of the format poolname/volname
+			 * into two separate strings.
+			 */
+			char str[256];
+			strcpy(str, zv->name);
```

Comment:
can we avoid string operations?
poolname can be obtained from spa and volname can be from main_zv
so, you do, zv->main_zv->zv_spa->spa_name for poolname,
and, zv->main_zv->zv_name for volname
Also, change the variable name 'zv' to 'zinfo' in this function

Reply:
ofcourse, you add all the null checks

Reply:
there is need of global lock here.
@pawanpraka1 can we reduce the need of lock for long time here?

Reply:
something like at uzfs_zvol_timer_thread

Reply:
@vishnuitta zv->main_zv->zv_name gives following output
``` 
      "volName": "pool1/vol1"
```